src/main/java/org/apache/maven/plugins/clean/CleanMojo.java

```diff
@@ -161,6 +161,12 @@
     @Parameter( property = "maven.clean.excludeDefaultDirectories", defaultValue = "false" )
     private boolean excludeDefaultDirectories;
 
+    @Parameter( property = "maven.clean.fast", defaultValue = "false" )
+    private boolean fast;
+
+    @Parameter( property = "maven.clean.fastDir", defaultValue = "${maven.multiModuleProjectDirectory}/.mvn/clean" )
```

Comment:
Why `~/.mvn/clean`? Would renaming `target/` to `target-XXX` not work?

Reply:
I was thinking that in case the background job is not finished and if we restart a clean goal, we'd want to get rid of those previous unfinished attempts, so I'd rather keep all those "to be deleted" files under a single directory, so that's why I originally ruled out a simple rename. That makes the logic a bit more complicated.

I'm currently experimenting with a way to make sure files are deleted when the session is ended (currently, maven exits with a `System.exit(1)` which cause the background thread to exit without having the time to delete the files.  This may make the above problem (of getting lots of "to be deleted" files that could not be deleted fast enough, hence a need to clean them again on a next run).  

Ideally, I think those files should be stored in `${maven.multiModuleProjectDirectory}/target/.clean`.

Reply:
* the defaults is now `${maven.multiModuleProjectDirectory}/target/.clean`
* the mojo waits until all files are deleted when the session is ended
* in case the `fastDir` is inside a `target` directory, the `target` directory is correctly deleted at the end